packages/adminrouter/extra/src/README.md

```diff
@@ -692,13 +718,229 @@ Tracking the chain of fixtures and how they use each other may provide better
 understanding of how the test harness works.
 
 ### JWT
-DC/OS IAM relies on JSON Web Tokens for transmitting security information between DC/OS
-components. Open repository flavour uses HS256 tokens, while EE relies on RS256.
-Test harness needs to have a way to create them for use in the tests. This is
-done by `test-harness/modules/mocker/jwt.py` module, which together with
-`repo_is_ee` fixture provides abstracts away the type of the
+DC/OS IAM relies on JSON Web Tokens for transmitting security information
+between DC/OS components. Open repository flavour uses HS256 tokens, while EE
+relies on RS256.  Test harness needs to have a way to create them for use in
+the tests. This is done by `test-harness/modules/mocker/jwt.py` module, which
+together with the `repo_is_ee` fixture, provides abstracts away the type of the
 token used by Admin Router itself.
 
+### Tests code reuse and deduplication
+Lots of endpoints exposed by Admin Router share common behaviour. Usually, it
+is something simple like testing for correct headers, the right upstream, etc...
+This results in tests covering them being very repetitive.
+
+In order to keep the code DRY, common code was extracted into a library with
+generic tests and helpers and a testing framework using YAML configuration files
+was developed.
+
+#### Generic tests
+Tests shared by most of the endpoints are as follows:
+* `generic_no_slash_redirect_test` - test if a request for a location without
```

Comment:
Some of these end with "." and some don't. Make them consistent (personally I prefer with a full stop).